templates/company-board/contracts/CompanyBoardTemplate.sol

```diff
@@ -38,97 +38,109 @@ contract CompanyBoardTemplate is BaseTemplate {
+            _shareHolders,
+            _shareStakes,
+            _shareVotingSettings
+        );
+
+        _setupPermissions(_dao, boardVoting, boardTokenManager, shareVoting, shareTokenManager, agentOrVault, finance, _useAgentAsVault);
+    }
+
+    function _installTokenApps(
+        Kernel _dao,
+        address[] _boardMembers,
+        uint64[3] _boardVotingSettings,
+        address[] _shareHolders,
+        uint256[] _shareStakes,
+        uint64[3] _shareVotingSettings
+    )
+        internal
+        returns (TokenManager boardTokenManager, Voting boardVoting, TokenManager shareTokenManager, Voting shareVoting)
+    {
         require(_boardMembers.length > 0, ERROR_MISSING_BOARD_MEMBERS);
-        (Kernel dao, MiniMeToken boardToken,) = _getCache(msg.sender);
-
-        // Install apps
-        ACL acl = ACL(dao.acl());
-        _boardTokenManager = _installTokenManagerApp(dao, boardToken, BOARD_TRANSFERABLE, BOARD_MAX_PER_ACCOUNT);
-        _boardVoting = _installVotingApp(dao, boardToken, _boardSupportRequired, _boardMinAcceptanceQuorum, _boardVoteDuration);
-
-        // Mint tokens
-        _mintTokens(acl, _boardTokenManager, _boardMembers, 1);
-    }
-
-    function _setupShare(address[] _shareHolders, uint256[] _shareStakes, uint64 _shareVoteDuration, uint64 _shareSupportRequired, uint64 _shareMinAcceptanceQuorum) internal returns (TokenManager _shareTokenManager, Voting _shareVoting) {
+        require(_boardVotingSettings.length == 3, ERROR_BAD_VOTE_SETTINGS);
         require(_shareHolders.length > 0, ERROR_MISSING_SHARE_MEMBERS);
         require(_shareHolders.length == _shareStakes.length, ERROR_BAD_HOLDERS_STAKES_LEN);
-        (Kernel dao,, MiniMeToken shareToken) = _getCache(msg.sender);
+        require(_shareVotingSettings.length == 3, ERROR_BAD_VOTE_SETTINGS);
+
+        ACL acl = ACL(_dao.acl());
+        (MiniMeToken boardToken, MiniMeToken shareToken) = _popTokensCache(msg.sender);
 
-        // Install apps
-        ACL acl = ACL(dao.acl());
-        _shareTokenManager = _installTokenManagerApp(dao, shareToken, SHARE_TRANSFERABLE, SHARE_MAX_PER_ACCOUNT);
-        _shareVoting = _installVotingApp(dao, shareToken, _shareSupportRequired, _shareMinAcceptanceQuorum, _shareVoteDuration);
+        boardVoting = _installVotingApp(_dao, boardToken, _boardVotingSettings[0], _boardVotingSettings[1], _boardVotingSettings[2]);
+        shareVoting = _installVotingApp(_dao, shareToken, _shareVotingSettings[0], _shareVotingSettings[1], _shareVotingSettings[2]);
+        boardTokenManager = _installTokenManagerApp(_dao, boardToken, BOARD_TRANSFERABLE, BOARD_MAX_PER_ACCOUNT);
+        shareTokenManager = _installTokenManagerApp(_dao, shareToken, SHARE_TRANSFERABLE, SHARE_MAX_PER_ACCOUNT);
 
-        // Mint tokens
-        _mintTokens(acl, _shareTokenManager, _shareHolders, _shareStakes);
+        _mintTokens(acl, boardTokenManager, _boardMembers, 1);
+        _mintTokens(acl, shareTokenManager, _shareHolders, _shareStakes);
```

Comment:
Not a huge thing, but I don't really see the point of `_setupApps()` and `_installTokenApps()` if they could just be rolled into `setupInstance()`

Reply:
I already tried that, we are getting `stack too deep` issues